core/domain/state_domain.py

```diff
@@ -2112,6 +2112,57 @@ def convert_html_fields_in_state(cls, state_dict, conversion_fn):
                             answer_group_index]['rule_specs'][
                                 rule_spec_index]['inputs']['x'][x_index] = (
                                     conversion_fn(x))
+            if state_dict['interaction']['id'] == 'DragAndDropSortInput':
+                for rule_spec_index, rule_spec in enumerate(
+                        answer_group['rule_specs']):
+                    if rule_spec['rule_type'] == 'HasElementXAtPositionY':
+                        x = state_dict['interaction']['answer_groups'][
+                            answer_group_index]['rule_specs'][
+                                rule_spec_index]['inputs']['x']
+                        state_dict['interaction']['answer_groups'][
+                            answer_group_index]['rule_specs'][
+                                rule_spec_index]['inputs']['x'] = (
+                                    conversion_fn(x))
+                    elif rule_spec['rule_type'] == 'IsEqualToOrdering':
+                        for x_index, x in enumerate(rule_spec['inputs']['x']):
+                            state_dict['interaction']['answer_groups'][
+                                answer_group_index]['rule_specs'][
+                                    rule_spec_index]['inputs']['x'][
+                                        x_index][0] = conversion_fn(x[0])
+                    elif rule_spec['rule_type'] == 'HasElementXBeforeElementY':
```

Comment:
I see this as a duplicate code, can we do something here?

Reply:
refactored. done